src/components/PopupWithForm.js

```diff
@@ -16,10 +18,19 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
+  setLoading(isLoading, savingText = "Saving...") {
+    if (isLoading) {
+      this._submitButton.textContent = savingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`